File: Coding Exercises/Python/Day16/sum_of_extremes.py
def sum_of_extremes(arr):
    if not arr:
        raise ValueError("Input list is empty")

    min_val = arr[0]
    max_val = arr[0]

    for num in arr[1:]:
        if num < min_val:
            min_val = num

        if num > max_val:
            max_val = num

    return min_val + max_val

File: Coding Exercises/TypeScript/Day16/sumOfExtremes.ts
function sumOfExtremes(arr: number[]): number {
  if (!arr || arr.length === 0) {
      throw new Error("Input array is empty");
  }

  let min = arr[0];
  let max = arr[0];

  for (let i = 1; i < arr.length; i++) {
      if (arr[i] < min) {
          min = arr[i];
      }

      if (arr[i] > max) {
          max = arr[i];
      }
  }

  return min + max;
}
